R/add_fcm_phase.R

```diff
@@ -1,59 +1,109 @@
-#
-#   df <- df %>% dplyr::mutate(fc_phase = ifelse(.data$fc_cell == 1 | .data$fc_cell == 6, "Phase 1 FC",
-#                                                ifelse(.data$fc_cell == 2 |.data$fc_cell == 3|.data$fc_cell == 7|.data$fc_cell == 11|.data$fc_cell == 12|.data$fc_cell == 16|.data$fc_cell == 17|.data$fc_cell == 18|.data$fc_cell == 21|.data$fc_cell == 22|.data$fc_cell == 26|.data$fc_cell == 31|.data$fc_cell == 32|.data$fc_cell == 36, "Phase 2 FC",
-#                                                       ifelse(.data$fc_cell == 4|.data$fc_cell == 5|.data$fc_cell == 8|.data$fc_cell == 9|.data$fc_cell == 13|.data$fc_cell == 19|.data$fc_cell == 20|.data$fc_cell == 23|.data$fc_cell == 24|.data$fc_cell == 27|.data$fc_cell == 28|.data$fc_cell == 33|.data$fc_cell == 34|.data$fc_cell == 37|.data$fc_cell == 38|.data$fc_cell == 41|.data$fc_cell == 42|.data$fc_cell == 43,"Phase 3 FC",
-#                                                              ifelse(.data$fc_cell == 10|.data$fc_cell == 14|.data$fc_cell == 15|.data$fc_cell == 25|.data$fc_cell == 29|.data$fc_cell == 35|.data$fc_cell == 39|.data$fc_cell == 40|.data$fc_cell == 44, "Phase 4 FC",
-#                                                                     ifelse(.data$fc_cell == 30 | .data$fc_cell == 45, "Phase 5 FC", NA))))))
-#
-# }
+
+#' Add the food consumption matrix to the dataset
+#'
+#'
+#' @param dataset A dataframe
+#' @param fews_vars A vector of 3 strings for the column names of fcs, hhs, and rcsi categories from the dataframe
+#' it has to include all "fcs_cat", "rcsi_cat" and "hhs_cat"
+#' @param fcs_categories A vector of 3 strings for the values included in the fcs_cat column
+#' It have to be a character and include "Acceptable", "Poor" and "Borderline"
+#' @param rcsi_categories A vector of 3 strings for the values included in the rcsi_cat column
+#' It have to be a character and include "No to Low", "Medium" and "Severe"
+#' @param hhs_categories A vector of 5 strings for the values included in the hhs_cat column
+#' It have to be a character and include "None","Little","Moderate"," Severe","Very Severe"
+#'
+#' @return this function returns a dataframe with a column called fc_cell that includes values from 1 to 45
+#' representing the Food Consumption Score Matrix and the fc_phase column that includes the different 5 phases
+#' of food consumption
+#' @export
+#' @importFrom dplyr case_when mutate
+add_fcm_phase <- function(dataset,
+                          fews_vars = c("fcs_cat","rcsi_cat","hhs_cat"),
+                          fcs_categories = c("Acceptable", "Poor", "Borderline"),
+                          rcsi_categories = c("No to Low", "Medium", "Severe"),
```

Comment:
This is my mistake since you built this off an older dataset, but I've been corrected by our FSL Specialist that the highest rcsi category should be "High", not "Severe". the add_rcsi code correctly reflects this, but it should be updated in this file throughout ("High", not "Severe" for rcsi)... otherwise thees functions won't speak to each other.